python/lsst/sims/maf/metrics/snSLMetric.py

```diff
@@ -156,38 +177,47 @@ def run(self, dataSlice, slicePoint=None):
 
         return N_lensed_SNe_Ia
 
-    def seasonCalc(self, obs):
+    def coadd(self, dataSlice):
         """
-        Method to estimate seasons
+        Method to coadd data per band and per night
 
         Parameters
-        --------------
-       obs: numpy array
-          array of observations
-        season_gap: float, opt
-          minimal gap required to define a season (default: 80 days)
-        mjdCol: str, opt
-          col name for MJD infos (default: observationStartMJD)
+        ---------------
+        dataSlice : np.array
 
         Returns
-        ----------
-        original numpy array sorted and season 
+        -----------
+        dataSlice, with m5Col replaced with m5 coadded in the night
 
         """
 
-        # check wether season has already been estimated
-        if 'season' in obs.dtype.names:
-            return obs, obs['season']
+        dataSlice.sort(order=[self.nightCol, self.filterCol])
+
+        filters = np.unique(dataSlice[self.filterCol])
+
+        for filtername in filters:
+            infilt = np.where(dataSlice[self.filterCol] == filtername)[0]
+            unight, indices = np.unique(dataSlice[self.nightCol][infilt], return_inverse=True)
+            right = unight+0.5
+            bins = [unight[0]-0.5] + right.tolist()
+            coadds, be, bn = binned_statistic(dataSlice[self.nightCol][infilt],
+                                              dataSlice[self.m5Col][infilt], bins=bins,
+                                              statistic=coaddM5)
+            # now we just clobber the m5Col with the coadd for the night
+            dataSlice[self.m5Col][infilt] = coadds[indices]
```

Comment:
Does this have to happen in the dataSlice or can it be stored elsewhere? I am anxious as to whether this is a view or a copy of the original dataslice (i.e. if we're running metrics in the same metricbundlegroup same 'constraint' set -- which receive the same dataslice at every healpix point -- are they getting the original m5 values or these clobbered values?)